traffic_ops/app/lib/API/Tenant.pm

```diff
@@ -0,0 +1,247 @@
+		return $self->alert("A tenant with name \"$name\" already exists.");
+	}
+
+	my $is_active = exists($params->{active})? $params->{active} : 0; #optional, if not set use default
+	
+	if ( !$is_active && !defined($parent_id)) {
+		return $self->alert("Root user cannot be in-active.");
+	}
+	
+	my $values = {
+		name 		=> $params->{name} ,
+		active		=> $is_active,
+		parent_id 	=> $params->{parentId}
+	};
+
+	my $insert = $self->db->resultset('Tenant')->create($values);
+	my $rs = $insert->insert();
+	if ($rs) {
+		my $response;
+		$response->{id}          	= $rs->id;
+		$response->{name}        	= $rs->name;
+		$response->{active}        	= \$rs->active;
+		$response->{parentId}           = $rs->parent_id;
+		$response->{lastUpdated} 	= $rs->last_updated;
+
+		&log( $self, "Created Tenant name '" . $rs->name . "' for id: " . $rs->id, "APICHANGE" );
+
+		return $self->success( $response, "Tenant create was successful." );
+	}
+	else {
+		return $self->alert("Tenant create failed.");
+	}
+
+}
+
+
+sub delete {
+	my $self = shift;
+	my $id     = $self->param('id');
+
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	my $tenant = $self->db->resultset('Tenant')->find( { id => $id } );
+	if ( !defined($tenant) ) {
+		return $self->not_found();
+	}	
+	my $name = $self->db->resultset('Tenant')->search( { id => $id } )->get_column('name')->single();
+	
+	my $existing_child = $self->db->resultset('Tenant')->search( { parent_id => $id } )->get_column('name')->first();
+	if ($existing_child) {
+		return $self->alert("Tenant '$name' has children tenant(s): e.g '$existing_child'. Please update these tenants and retry.");
+	}
+
+	#The order of the below tests is intentional - allowing UT to cover all cases - TODO(nirs) remove this comment when a full "tenancy" UT is added, including permissions and such (no use in putting effort into it yet)
+	#TODO(nirs) - add back when making available: my $existing_ds = $self->db->resultset('Deliveryservice')->search( { tenant_id => $id })->get_column('xml_id')->first();
+	#TODO(nirs) - add back when making available: if ($existing_ds) {
+	#TODO(nirs) - add back when making available: 	return $self->alert("Tenant '$name' is assign with delivery-services(s): e.g. '$existing_ds'. Please update/delete these delivery-services and retry.");
+	#TODO(nirs) - add back when making available: }
+
+	#TODO(nirs) - add back when making available: my $existing_cdn = $self->db->resultset('Cdn')->search( { tenant_id => $id })->get_column('name')->first();
+	#TODO(nirs) - add back when making available: if ($existing_cdn) {
+	#TODO(nirs) - add back when making available: 	return $self->alert("Tenant '$name' is assign with CDNs(s): e.g. '$existing_cdn'. Please update/delete these CDNs and retry.");
+	#TODO(nirs) - add back when making available: }
+
+	#TODO(nirs) - add back when making available: my $existing_user = $self->db->resultset('TmUser')->search( { tenant_id => $id })->get_column('username')->first();
```

Comment:
No Problem. 
Note that you'll need to be careful on the merge when the next 3 PRs are added - as the order of the tests is important for the UT, I hope to fix it later on.